src/Microsoft.Docs.Template/Models/Attributes.cs

```diff
@@ -9,35 +9,47 @@ namespace Microsoft.Docs.Build
     /// Exports a type to be processed by build pipeline into a JSON model.
     /// </summary>
     [AttributeUsage(AttributeTargets.Class)]
-    public class DataSchemaAttribute : Attribute{ }
+    public class DataSchemaAttribute : Attribute { }
 
     /// <summary>
     /// Exports a type to be processed by build pipeline into an HTML page.
     /// </summary>
     [AttributeUsage(AttributeTargets.Class)]
-    public class PageSchemaAttribute : DataSchemaAttribute { }
-
-    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
-    public abstract class DataTypeAttribute : Attribute
+    public class PageSchemaAttribute : DataSchemaAttribute
     {
-        public Type RequiredType { get; set; }
-    }
+        public bool Contributors { get; }
 
-    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
-    public class HrefAttribute : DataTypeAttribute
-    {
-        public HrefAttribute() => RequiredType = typeof(string);
-    }
+        public bool GitUrl { get; }
 
-    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
-    public class MarkdownAttribute : DataTypeAttribute
-    {
-        public MarkdownAttribute() => RequiredType = typeof(string);
+        public bool DocumentId { get; }
+
+        public bool Toc { get; }
+
+        public PageSchemaAttribute(
+            bool contributors = true,
+            bool gitUrl = true,
+            bool documentId = true,
+            bool toc = true)
```

Comment:
change names to more like bool? maybe "enableXXX"...

Reply:
yeah

---
In reply to: [213596931](https://github.com/dotnet/docfx/pull/3316#discussion_r213596931) [](ancestors = 213596931)